models/tag.js

```diff
@@ -20,9 +26,15 @@ const TagSchema = new mongoose.Schema({
 });
 
 TagSchema.pre('save', function(next) {
-  const slug = this.name.replace(/\s/g, '').toLowerCase();
+  const slug = this.name.replace(/\s+/g,'-').toLowerCase();
   this.slug = slug;
   next();
 });
 
+TagSchema.pre(['updateOne','findOneAndUpdate'],function(next){
```

Comment:
Remove this

Reply:
without this the pre save thing in model doesnt get updated on findOneAndUpdate .

Reply:
I mean slug doesnt updates on findOneAndUpdate @vixrant